src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -0,0 +1,228 @@
+    private val party = mutableListOf<PartyMember>()
+
+    //0 = not awaiting, 1 = awaiting 2nd delimiter, 2 = awaiting 1st delimiter
+    private var awaitingMessage = 0
+
+    @SubscribeEvent
+    fun onCommandRun(event: SendChatMessageEvent) {
+        if (!Utils.inSkyblock || !Skytils.config.partyAddons) return
+        if (Utils.equalsOneOf(event.message, "/pl", "/party list", "/p list", "/party l")) {
+            awaitingMessage = 2
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: ClientChatReceivedEvent) {
+        if (!Utils.isOnHypixel || event.type == 2.toByte() || !Skytils.config.partyAddons) return
+        val message = event.message.formattedText
+
+        if (message == "§f§r" && awaitingMessage != 0) {
+            event.isCanceled = true
+        } else if (partyStartPattern.matches(message)) {
+            party.clear()
+            event.isCanceled = true
+        } else if (message.startsWith("§eParty Leader")) {
+            val leader = leaderPattern.find(message)
+
+            if (leader != null) {
+                val name = leader.groups["name"]?.value!!
+                val status = leader.groups["status"]?.value!!
+                val rank = leader.groups["rank"]?.value!!
+
+                party.add(PartyMember(name, PartyMemberType.LEADER, status, rank))
+                event.isCanceled = true
+            }
+        } else if (message.startsWith("§eParty Moderators: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Moderators: ")).forEach {
+                val name = it.groups["name"]?.value!!
+                val status = it.groups["status"]?.value!!
+                val rank = it.groups["rank"]?.value!!
+
+                party.add(
+                    PartyMember(
+                        name,
+                        PartyMemberType.MODERATOR,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§eParty Members: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Members: ")).forEach {
+                val name = it.groups["name"]?.value ?: return
+                val status = it.groups["status"]?.value ?: return
+                val rank = it.groups["rank"]?.value ?: return
+
+                party.add(
+                    PartyMember(
+                        name,
+                        PartyMemberType.MEMBER,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§cYou are not currently in a party.") && awaitingMessage != 0) {
+            party.clear()
+        } else if (message == "§9§m-----------------------------------------------------§r" && awaitingMessage != 0) {
```

Comment:
Maybe Lily has other opinions about this but we could probably avoid hard coding the message and instead do a contains check for a lot of hyphens. Just in case Hypixel decides to changes things up for no reason.

Reply:
startswith maybe? unless you want a regex